tests/unit-tests/region-owner.cxx

```diff
@@ -0,0 +1,42 @@
+#include "doctest/doctest.h"
+
+#include <ipr/traversal>
+#include <ipr/impl>
+#include <ipr/utility>
+
+const ipr::Type* get_enclosing_udt_if_any(const ipr::Region& current_region)
+{
+    auto* region = &current_region;
+    for (;;)
+    {
+        switch (region->owner().category)
+        {
+        case ipr::Category_code::Class:
+        case ipr::Category_code::Enum:
+        case ipr::Category_code::Union:
+            return ipr::util::view<ipr::Type>(region->owner());
+

```

Comment:
Even though `category` is part of the _interface_ file, I strongly discourage taking strong dependency on it.  Use visitors.